src/ObjCRuntime/Registrar.cs

```diff
@@ -186,7 +186,7 @@ public void VerifyRegisterAttribute (ref List<Exception> exceptions)
 				}
 			}
 
-			static char[] invalidSelectorCharacters = new char[] { ' ', '\t' };
+			static char[] invalidSelectorCharacters = new char[] { ' ', '\t', '?', '\\', '!', '|', '@', '"', '\'', '%', '&', '/', '(', ')', '=', '^', '[', ']', '{', '}', ',', '.', ';', '-', '\n' };
```

Comment:
The thing is that there's no official Objective-C specification, and this list of characters was created manually (by trying it out and see if it failed to build or not).

It's not meant to be a comprehensive list, just a best effort to help customers a little bit, so I'd prefer to not put this in our public API.

The list also varies depending on whether you're compiling Objective-C++ or Objective-C (Objective-C allows characters Objective-C++ doesn't).